src/CMakeLists.txt

```diff
@@ -231,6 +226,12 @@ if (HYPRE_WITH_CUDA)
   # Check if CUDA is available, then enable it
   include(CheckLanguage)
   check_language(CUDA)
+
```

Comment:
@osborn9 "For CUDA which can have an explicit host compiler, the cache `CMAKE_CUDA_HOST_COMPILER` variable will be set if it was required for compilation (and cleared if it was not)." It could be whether `check_language(CUDA)` decide whether `CMAKE_CUDA_HOST_COMPILER` is "required" can be system dependent.
In any case, it's not good to rely on `check_language(CUDA)` to decide whether `CMAKE_CUDA_HOST_COMPILER` is necessary if you know you need it.

Reply:
Thanks @liruipeng for the explanation / improvement.